pkg/f1/run/test_runner.go

```diff
@@ -223,10 +233,12 @@ func (r *Run) doWork(doWorkChannel chan<- int32, durationElapsed *testing.Cancel
 		return
 	}
 	iteration := atomic.AddInt32(&r.iteration, 1)
-	if r.Options.MaxIterations > 0 && iteration == r.Options.MaxIterations {
-		doWorkChannel <- iteration
+	if r.Options.MaxIterations > 0 && iteration > r.Options.MaxIterations {
+		trace.Event("Max iterations exceeded Calling Cancel on iteration  '%v' .", iteration)
 		durationElapsed.Cancel()
```

Comment:
maybe print a log message to explain we're stopping due to max iterations, like we do for max duration?

Reply:
Fixed in ad24e3c8f258824ea21b1b7e2c1553954b84d891